lib/omniauth/strategies/shopify.rb

```diff
@@ -49,7 +49,7 @@ def valid_signature?
       def valid_scope?(token)
         params = options.authorize_params.merge(options_for("authorize"))
         return false unless token && params[:scope] && token['scope']
-        (params[:scope].split(',').sort == token['scope'].split(',').sort)
+        (params[:scope].delete(' ').split(',').sort == token['scope'].delete(' ').split(',').sort)
```

Comment:
It seems like `token['scope']` should never have any spaces.  However, could it be that `params[:scope]` is the one with the spaces that is causing the mismatch?